File: request/request.py

      
       
    # user:pass socks5
    import requests

    response = requests.get(
        "http://ipinfo.io/",
        proxies={
            "http": "socks5://user:pass@ip:port",
            "https": "socks5://user:pass@ip:port",
        },
    )
    print(response.text)

    # user:pass http

    import requests

    response = requests.get(
        "http://ipinfo.io/",
        proxies={
            "http": "http://user:pass@ip:port",
            "https": "http://user:pass@ip:port",
        },
    )
    print(response.text)

          
    

File: request/request.go

      
       
    package main

    import (
    	"log"

    	"github.com/go-resty/resty/v2"
    )

    func main() {
    	// user:pass socks5
    	client := resty.New()
    	client.SetProxy("socks5://user:pass@ip:port")
    	response, err := client.R().Get("http://ipinfo.io/json")
    	if err != nil {
    		panic(err)
    	}
    	log.Println(string(response.Body()))

    	// user:pass http
    	client = resty.New()
    	client.SetProxy("http://user:pass@ip:port")
    	response, err = client.R().Get("http://ipinfo.io/json")
    	if err != nil {
    		panic(err)
    	}
    	log.Println(string(response.Body()))
    }

          
    
